tls_codec/src/quic_vec.rs

```diff
@@ -6,15 +6,26 @@
 //! Note that we require, as the MLS specification does, that vectors have to
 //! use the minimum number of bytes necessary for the encoding.
 //! This ensures that encodings are unique.
-
+//!
+//! With the `mls` feature the length of variable length vectors can be limited
+//! to 30-bit values.
+//! This is in contrast to the default behaviour defined by RFC 9000 that allows
+//! up to 62-bit length values.
 use alloc::vec::Vec;
 
 #[cfg(feature = "serde_serialize")]
 use serde::{Deserialize as SerdeDeserialize, Serialize as SerdeSerialize};
 
 use crate::{Deserialize, Error, Serialize, Size};
 
+#[cfg(not(feature = "mls"))]
 const MAX_LEN: u64 = 0x3fff_ffff_ffff_ffff; // <= (1<<62)-1
+#[cfg(not(feature = "mls"))]
+const MAX_LEN_LEN: usize = 8;
+#[cfg(feature = "mls")]
+const MAX_LEN: u64 = 0x3fff_ffff; // <= (1<<30)-1
```

Comment:
For clarity, I would use the expression from the comments: `(1 << 30) - 1`

The same also applies to the original length obviously.